src/ImageSharp/Memory/Allocators/MemoryAllocatorSettings.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Copyright (c) Six Labors.
+// Licensed under the Apache License, Version 2.0.
+
+namespace SixLabors.ImageSharp.Memory
+{
+    /// <summary>
+    /// Defines options for creating the default <see cref="MemoryAllocator"/>.
+    /// </summary>
+    public struct MemoryAllocatorSettings
```

Comment:
I named this `MemoryAllocatorSettings` instead of a more conventional name candidate `MemoryAllocatorOptions`, so it's not confused with `MemoryAllocationOptions`.

Reply:
> You mean `AllocationOptions` yeah ?

Yeah, I meant that. It would be still somewhat confusing.

I'm not sure if `BufferInitializationBehavior` is a future-proof name. We can potentially introduce flags for alignment constraints or similar stuff.

Reply:
`BufferInitializationOptions` then. I think that would be enough to allow us to keep `MemoryAllocationOptions` and allow future-proofing. I'd really like to maintain the convention on upper API terminology.

Reply:
If we update the naming I'm more than happy for this to get merged in.

Reply:
I don't like `BufferInitializationOptions` since it's really not only about initialization. I think `AllocationOptions` describes the purpose perfectly, I'd prefer to keep that name.

I would rather rename `MemoryAllocatorSettings` to `MemoryAllocatorCreationOptions`. A bit verbose, but I can't come up with a better compromise.

Reply:
I'm curious about these potential future flags? We'd have to change the type from an enum to accommodate them. 

I think we should go with `MemoryAllocationOptions` at the upper level and make any compromise with  `AllocationOptions`. I don't actually agreed that the latter describes the purpose well. It's very abstract.

Reply:
> I'm curious about these potential future flags? 

For example

```C#
using var mem = memoryAllocator.Allocate<byte>(AllocationOptions.Aligned32);
```

would allocate a buffer where the starting address is divisible by 32 (aligned to 32 byte boundaries). This results in faster AVX loads.

> I don't actually agreed that the latter describes the purpose well. It's very abstract.

I don't see why, it defines flags customizing the `Allocate` call. I think `MemoryAllocatorOptions` / `MemoryAllocatorCreationOptions` is a very narrow customer-facing use case, while `AllocationOptions` is everywhere in our codebase, heavily used by community contributors, I don't like the idea of forcing an unnatural name there.

Reply:
Ahhhh right! You meant literal flags! Your example clears everything up there. `AllocationOptions` is fine. Sorry 🙏

`Creation` feels like an odd word then, not a computing term. I think we should just go with `MemoryAllocatorOptions` if the use case is so narrow. You'd have to be coding with Notepad not to see the difference.

Reply:
> `Creation` feels like an odd word then, not a computing term.

I think [I get what you mean](https://www.google.com/search?q=creation&tbm=isch) 😆. Let's keep it `MemoryAllocatorOptions` then!